roles/openshift_on_openstack/tasks/main.yml

```diff
@@ -188,50 +201,84 @@
     dest: "{{ osev3_yml_path }}"
     backup: yes
 
-
-# Get some debug information that makes it easier to file bugs.
-- name: Running commands that create debug information needed to open issues
-  shell: "{{ item }}"
-  register: debug_information
-  with_items:
-    - "printf \"$(uname -srm)\n$(cat /etc/os-release)\n\""
-    - "ansible --version"
-    - "cd openshift-ansible ; git describe"
-    - "rpm -q atomic-openshift-utils openshift-ansible"
-  changed_when: false
+- name: Copy {{ osev3_yml_path }} to the artifacts directory
+  synchronize:
+    src: "{{ osev3_yml_path }}"
+    dest: "{{ artifacts_directory }}"
+    mode: pull
   ignore_errors: true
 
-# Print the debug information out.
-- name: Printing the debug information to the screen
-  debug:
-    msg: "{{ item['stdout_lines'] }}"
-  with_items: "{{ debug_information['results'] }}"
-
 # Set the path for the prerequisite log file.
 - name: Creating a variable for the prerequisite log file
   set_fact:
     prerequisite_log: "{{ ansible_user_dir }}/openshift_prerequisites.log"
 
-# Run the Ansible playbook that checks the prerequisites in OpenStack.
-- name: Checking the prerequisites for cluster provisioning in OpenStack
-  # Using two inventory flags here as the README instructs.
-  shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} {{ openshift_openstack_dir }}/openshift-cluster/prerequisites.yml 2>&1 >> {{ prerequisite_log }}"
-  args:
-    # Use bash to get the posix style redirects.
-    executable: /bin/bash
+- name: OpenShift prerequisite block
+  block:
+    # Run the Ansible playbook that checks the prerequisites in OpenStack.
+    - name: Checking the prerequisites for cluster provisioning in OpenStack
+      # Using two inventory flags here as the README instructs.
+      shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} {{ openshift_openstack_dir }}/openshift-cluster/prerequisites.yml 2>&1 >> {{ prerequisite_log }}"
+      args:
+        # Use bash to get the posix style redirects.
+        executable: /bin/bash
+  rescue:
+    # Get some debug information that makes it easier to file OpenShift bugs.
+    - name: Running commands that create debug information needed to open issues
+      shell: "{{ item }}"
```

Comment:
You can run a few commands as a single statement with shell and dump the output into a local "debug log" directory.  That would simplify this as well as keep the information both on the server running this inside it's artifacts directory and the local debug directory which will prove useful when actively debugging.